src/components/AdminSettings/GeneralSettings.vue

```diff
@@ -86,10 +114,28 @@ export default {
 		this.loading = true
 		this.defaultGroupNotification = defaultGroupNotificationOptions[parseInt(loadState('spreed', 'default_group_notification')) - 1]
 		this.loading = false
+
+		const signaling = loadState('spreed', 'signaling_servers')
+		this.updateSignalingServers(signaling.servers)
+		EventBus.on('signaling-servers-updated', this.updateSignalingServers)
+	},
+
+	beforeDestroy() {
+		EventBus.off('signaling-servers-updated', this.updateSignalingServers)
 	},
 
 	methods: {
 		t,
+
+		updateSignalingServers(servers) {
+			this.canEnableE2EE = servers.length > 0
```

Comment:
gets kind of tricky, but basically we need to additionally check the HPBs welcome page result and it must not be missing the feature `join-features`

Reply:
As discussed, the feature `join-features` is only necessary with the current implementation in #14005. I will change this to evaluate the global feature / appconfig from here.